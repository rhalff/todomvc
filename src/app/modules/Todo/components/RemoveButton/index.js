// @flow
import React from 'react'
import styles from './scss/RemoveButton.scss'

type Props = {
  onClick: () => void
}

const RemoveButton = ({ onClick }: Props) => (
  <button className={styles.removeButton} onClick={onClick}>
    Ã—
  </button>
)

export default RemoveButton
