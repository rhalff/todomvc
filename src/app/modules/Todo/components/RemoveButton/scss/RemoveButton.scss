@import 'vars';

.removeButton {
  width: 40px;
  height: 40px;
  margin: auto 0;
  font-size: 30px;
  color: $removeButtonColor;
  transition: color 0.2s ease-out;

  /*
  &:after {
    content: 'Ã—';
  }
  */

  &:hover {
    color: $removeButtonHoverColor;
  }
}

