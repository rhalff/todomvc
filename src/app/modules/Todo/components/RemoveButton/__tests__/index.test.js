// @flow
import React from 'react'
import RemoveButton from '../'
import renderer from 'react-test-renderer'

test('Can show RemoveButton', () => {
  const component = renderer.create(<RemoveButton onChange={() => {}} />)

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Can show RemoveButton (checked)', () => {
  const component = renderer.create(
    <RemoveButton onChange={() => {}} checked />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
