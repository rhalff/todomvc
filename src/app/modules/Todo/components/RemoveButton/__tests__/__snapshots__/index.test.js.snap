// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show RemoveButton (checked) 1`] = `
<button
  className="removeButton"
  onClick={undefined}
>
  ×
</button>
`;

exports[`Can show RemoveButton 1`] = `
<button
  className="removeButton"
  onClick={undefined}
>
  ×
</button>
`;
